03/app.js

```diff
@@ -1,25 +1,66 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
+import React from "react";
+import { createRoot } from "react-dom/client";
+import Category from "./Category";
+import Cart from "./Cart";
+import Product from "./Product";
+import data from "./data.json";
 
-import Category from './Category';
-import Cart from './Cart';
+class App extends React.Component {
+	state = {
+		cart: [],
+	};
 
-import data from './data.json';
+	addToCart = id => {
+		const product = data.find(item => item.id === id);
+		this.setState({ cart: [...this.state.cart, product] });
+	};
 
-class App extends React.Component {
-    state = {
-        cart: [],
-    }
-    
-    render() {
-        return (
-            <section>
-                <Category />
-                <Cart />
-            </section>
-        )
-    }
+	checkIfInCart = id => {
+		const product = this.state.cart.find(item => item.id === id);
+
+		if (this.state.cart.includes(product)) {
+			return true;
+		}
+		return false;
```

Comment:
Tutaj zdecydowanie da się prościej. Jeśli w `product` nie jest `null` to już to oznacza, że ten element jest w `cart`